src/main/java/mServer/crawler/sender/br/BrFilmDeserializer.java

```diff
@@ -0,0 +1,393 @@
+      }
+
+      if (!aDetailClip.has(JSON_ELEMENT_DURATION)) {
+        printMissingDetails(JSON_ELEMENT_DURATION);
+      }
+    }
+    return Optional.empty();
+  }
+
+  private Set<BrUrlDTO> edgesToUrls(final JsonObject viewer) {
+    final Set<BrUrlDTO> urls = new HashSet<>();
+    final Optional<JsonArray> edges = getVideoFileEdges(viewer);
+    if (edges.isPresent()) {
+      for (final JsonElement edge : edges.get()) {
+        final JsonObject ebdgeObj = edge.getAsJsonObject();
+        if (ebdgeObj.has(JSON_ELEMENT_NODE)) {
+          final JsonObject node = ebdgeObj.getAsJsonObject(JSON_ELEMENT_NODE);
+          final Optional<BrUrlDTO> url = nodeToUrl(node);
+          if (url.isPresent()) {
+            urls.add(url.get());
+          }
+        }
+      }
+    }
+    return urls;
+  }
+
+  private Optional<JsonElement> getBroadcastStart(final JsonObject aDetailClip) {
+    if (!aDetailClip.has(JSON_ELEMENT_BROADCASTS)) {
+      return Optional.empty();
+    }
+
+    final JsonObject broadcast = aDetailClip.getAsJsonObject(JSON_ELEMENT_BROADCASTS);
+    if (!broadcast.has(JSON_ELEMENT_EDGES)) {
+      return Optional.empty();
+    }
+
+    final JsonArray edges = broadcast.getAsJsonArray(JSON_ELEMENT_EDGES);
+    if (edges.size() <= 0) {
+      return Optional.empty();
+    }
+    
+    final JsonObject arrayItem = edges.get(0).getAsJsonObject();
+    if (!arrayItem.has(JSON_ELEMENT_NODE)) {
+      return Optional.empty();
+    }
+    
+    final JsonObject node = arrayItem.getAsJsonObject(JSON_ELEMENT_NODE);
+    if (!node.has(JSON_ELEMENT_START)) {
+      return Optional.empty();
+    }
+
+    return Optional.of(node.get(JSON_ELEMENT_START));
+  }
+
+  private Optional<JsonObject> getDetailClip(final JsonObject aViewer) {
+    if (!aViewer.has(JSON_ELEMENT_DETAIL_CLIP)) {
+      return Optional.empty();
+    }
+
+    return Optional.of(aViewer.getAsJsonObject(JSON_ELEMENT_DETAIL_CLIP));
+  }
+
+  private Optional<JsonArray> getVideoFileEdges(final JsonObject aViewer) {
+    if (!aViewer.has(JSON_ELEMENT_CLIP)) {
+      return Optional.empty();
+    }
+
+    final JsonObject clip = aViewer.getAsJsonObject(JSON_ELEMENT_CLIP);
+    if (!clip.has(JSON_ELEMENT_VIDEO_FILES)) {
+      return Optional.empty();
+    }
+
+    final JsonObject videoFiles = clip.getAsJsonObject(JSON_ELEMENT_VIDEO_FILES);
+    if (!videoFiles.has(JSON_ELEMENT_EDGES)) {
+      return Optional.empty();
+    }
+
+    return Optional.of(videoFiles.getAsJsonArray(JSON_ELEMENT_EDGES));
+  }
+
+  private Optional<JsonObject> getViewer(final JsonObject aBaseObject) {
+    if (!aBaseObject.has(JSON_ELEMENT_DATA)) {
+      return Optional.empty();
+    }
+
+    final JsonObject data = aBaseObject.getAsJsonObject(JSON_ELEMENT_DATA);
+    if (!data.has(JSON_ELEMENT_VIEWER)) {
+      return Optional.empty();
+    }
+
+    return Optional.of(data.getAsJsonObject(JSON_ELEMENT_VIEWER));
+  }
+
+  private Optional<BrUrlDTO> nodeToUrl(final JsonObject aNode) {
```

Comment:
![CRITICAL](https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL') Refactor this method to reduce its Cognitive Complexity from 20 to the 15 allowed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3776)